File: src/main/java/random/telegramhomebot/bootstrap/CommandsLoader.java
package random.telegramhomebot.bootstrap;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;
import random.telegramhomebot.model.TelegramCommand;
import random.telegramhomebot.repository.TelegramCommandRepository;

import java.util.Map;
import java.util.stream.Collectors;

@Slf4j
@RequiredArgsConstructor
@Component
public class CommandsLoader implements CommandLineRunner {

	private final Map<String, String> telegramCommands;
	private final TelegramCommandRepository telegramCommandRepository;

	@Override
	public void run(String... args) {
		loadSampleCommands();
	}

	private void loadSampleCommands() {
		long commandsCount = telegramCommandRepository.count();
		log.info("Stored commands count [{}]", commandsCount);
		if (commandsCount == 0) {
			log.info("Loading sample commands...");
			telegramCommandRepository.saveAll(telegramCommands.entrySet().stream()
					.map(entry -> new TelegramCommand(entry.getKey(), entry.getValue()))
					.collect(Collectors.toList()));
		}
	}
}

File: src/main/java/random/telegramhomebot/bootstrap/CommandsLoader.kt
package random.telegramhomebot.bootstrap

import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.boot.CommandLineRunner
import org.springframework.core.annotation.Order
import org.springframework.stereotype.Component
import random.telegramhomebot.model.TelegramCommand
import random.telegramhomebot.repository.TelegramCommandRepository
import random.telegramhomebot.utils.logger

@Order(2)
@Component
class CommandsLoader(
    private val telegramCommandRepository: TelegramCommandRepository,
    @Qualifier("telegramCommands")
    private val telegramCommands: Map<String, String>
) : CommandLineRunner {
    val log = logger()

    override fun run(vararg args: String) {
        log.info("CommandsLoader started")
        loadSampleCommands()
    }

    private fun loadSampleCommands() {
        val commandsCount = telegramCommandRepository.count()
        log.info("Stored commands count [{}]", commandsCount)
        if (commandsCount == 0L) {
            log.info("Loading sample commands...")
            telegramCommandRepository.saveAll(
                telegramCommands.entries.map { TelegramCommand(it.key, it.value, true) }
            )
        }
    }
}
